### PR TITLE
Fix custom popover padding

### LINKED ISSUE #1937: New popover padding breaks popover header
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

During our recent work on popover redesign (see PR #1803), we had to add some defensive CSS to prevent the potential edge case when a `.popover-header` without content is not displayed correctly ; see comment https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1803#discussion_r1145791920.

The chosen solution was to move the padding to the `.popover` itself instead of having a `.popover-header` and `.popover-body` padding and/or margin.

Unfortunately, the design is now broken for the custom example:
broken_popover

instead of:
popover

### Reduced test cases

To do:

- [ ] Fix custom example (by going back to old margin and/or padding?)
- [ ] Find another way to prevent the potential edge case when a `.popover-header` without content is not displayed correctly

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Boosted are you using?

v5.3.0

### PR DESCRIPTION
**Pull request by MewenLeHo**
### Related issues
 
 Closes #1937 
 
 ### Description
 
 Fix padding for custom popover while keeping new design unchanged.
 
 ### Motivation & Context
 
 Design broken after design update:
 229475553-11144aae-0982-4cee-ac21-b75d58beee85
 
 ### Types of change
 
 - Bug fix (non-breaking which fixes an issue)
 
 ### Live previews
 
 - <https://deploy-preview-2031--boosted.netlify.app/docs/5.3/components/popovers/#custom-popovers>
 
 ### Checklist
 
 
 
 
 #### Contribution
 
 - [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)
 
 #### Accessibility
 
 - [x] My change follows accessibility good practices; I have at least run axe
 
 #### Design
 
 - [x] My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
 - [x] My change is compatible with responsive display
 
 #### Development
 
 - [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
 - (NA) I have added JavaScript unit tests to cover my changes
 - (NA) I have added SCSS unit tests to cover my changes
 
 #### Documentation
 
 - [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly
 
 ### Checklist (for Core Team only)
 
 - (NA) My change introduces changes to the migration guide
 - (NA) My new component is added in Storybook
 - (NA) My new component is compatible with RTL
 - [x] Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
 - [x] Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
 - [ ] Code review
 - [ ] Design review
 - (NA) A11y review
 
 #### After the merge
 
 - [ ] Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

**Comment by julien-deramond** [comment]
_(macOS, Firefox)_

Rendering in the main branch:
Screenshot 2023-05-24 at 09 15 39

Rendering in this branch:
Screenshot 2023-05-24 at 09 16 09

There's an extra spacing between the title and the description. Is that normal?

**Comment by julien-deramond** [comment]
There's probably an issue with `--bs-popover-body-padding-y` which only changes the bottom padding instead of both bottom and top padding. Here's the rendering with `--bs-popover-body-padding-x: 3rem` and `--bs-popover-body-padding-y: 5rem`:


Screenshot 2023-05-31 at 09 08 00

**Comment by MewenLeHo** [comment]
> There's probably an issue with `--bs-popover-body-padding-y` which only changes the bottom padding instead of both bottom and top padding. Here's the rendering with `--bs-popover-body-padding-x: 3rem` and `--bs-popover-body-padding-y: 5rem`:
> 
> Screenshot 2023-05-31 at 09 08 00

I moved a CSS property and it should be better now:
screenshot-localhost_9001-2023 05 31-10_34_08

**Comment by isabellechanclou** [comment]
LGTM !

**Comment by julien-deramond** [comment]
Screenshot 2023-06-05 at 13 08 37

Is it possible with the custom props to center the text in the header? Doesn't seem very usable right now.

**Comment by MewenLeHo** [comment]
I removed specific some specific CSS rules for custom popovers so it should be better now.
I am just wondering if it is really a good think to add examples rules for popover header customisation in a Boosted mod? 🤔 

screenshot-deploy-preview-2031--boosted netlify app-2023 06 06-10_43_55

**Comment by isabellechanclou** [comment]
LGTM !

File: scss/_popover.scss [hunk 1]
```diff
@@ -203,15 +202,18 @@
...
 }
 
 .popover-body {
-  // Boosted mod: no `padding`
+  padding: var(--#{$prefix}popover-body-padding-y) var(--#{$prefix}popover-body-padding-x);
   color: var(--#{$prefix}popover-body-color);
 }
+
+// Boosted mod: needed to reset styles for custom popover and make it customisable
+.popover:not(.custom-popover) .popover-header + .popover-body {
```
Comment (julien-deramond) [inline scss/_popover.scss:213]: It's not possible to do that. `.custom-popover` comes from the documentation as an example, it doesn't exist in the library. One can use whatever name as a class.

File: scss/_variables.scss [hunk 2]
```diff
@@ -1789,14 +1789,11 @@ $popover-header-font-size:          $font-size-lg !default; // Boosted mod: inst
 $popover-header-line-height:        1.11 !default; // Boosted mod
 $popover-header-bg:                 $popover-bg !default; // Boosted mod: instead of `var(--#{$prefix}secondary-bg)`
 $popover-header-color:              $headings-color !default;
-$popover-header-padding-bottom:     map-get($spacers, 2) !default; // Boosted mod
-// fusv-disable
-$popover-header-padding-y:          $spacer * .5 !default; // Boosted mod: deprecated in v5.3.0 replaced by `$popover-header-padding-top` and `$popover-header-padding-bottom`
-// fusv-enable
+$popover-header-padding-y:          $spacer !default; // Boosted mod: instead of `.5rem`
```
Comment (julien-deramond) [inline scss/_variables.scss:1792]: Must be removed from the migration guide if reintroduced

File: scss/_popover.scss (hunk 1 above)
Comment (julien-deramond) [reply scss/_popover.scss:213]: @MewenLeHo Tried something in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2077

**Comment by MewenLeHo** [comment]
I just added to this PR the prototype made by @julien-deramond in PR #2077 
@isabellechanclou and @julien-deramond, could you please have a look on this new verson?

**Review by julien-deramond — APPROVED** [review]

File: scss/_popover.scss [hunk 3]
```diff
@@ -203,15 +209,19 @@
...
+  padding:
+    var(--#{$prefix}popover-body-padding-y, var(--#{$prefix}popover-body-padding-top))
+    var(--#{$prefix}popover-body-padding-x)
+    var(--#{$prefix}popover-body-padding-y, var(--#{$prefix}popover-body-padding-bottom))
+    var(--#{$prefix}popover-body-padding-x); // Boosted mod: instead of padding: `var(--#{$prefix}popover-body-padding-y) var(--#{$prefix}popover-body-padding-x);`
   color: var(--#{$prefix}popover-body-color);
 }
+
+// Boosted mod
+:not(.popover-header) + .popover-body {
```
Comment (louismaximepiton) [inline scss/_popover.scss:224]: I think the selector could be more specific to select popover-body here

```suggestion
.popover-arrow + .popover-body {
```

File: scss/_popover.scss [hunk 4]
```diff
@@ -203,15 +209,19 @@
...
-  // End mod
 }
 
 .popover-body {
-  // Boosted mod: no `padding`
+  padding:
+    var(--#{$prefix}popover-body-padding-y, var(--#{$prefix}popover-body-padding-top))
+    var(--#{$prefix}popover-body-padding-x)
+    var(--#{$prefix}popover-body-padding-y, var(--#{$prefix}popover-body-padding-bottom))
+    var(--#{$prefix}popover-body-padding-x); // Boosted mod: instead of padding: `var(--#{$prefix}popover-body-padding-y) var(--#{$prefix}popover-body-padding-x);`
```
Comment (louismaximepiton) [inline scss/_popover.scss:219]: Maybe it should be on one line ? Or don't we care ?

Should we explain this behavior in the doc/migration ?

Anyway proposal to remove second call to `var(--#{$prefix}popover-body-padding-x)`. Same above ?

```suggestion
  padding:
    var(--#{$prefix}popover-body-padding-y, var(--#{$prefix}popover-body-padding-top))
    var(--#{$prefix}popover-body-padding-x)
    var(--#{$prefix}popover-body-padding-y, var(--#{$prefix}popover-body-padding-bottom)); // Boosted mod: instead of padding: `var(--#{$prefix}popover-body-padding-y) var(--#{$prefix}popover-body-padding-x);`
```

File: scss/_popover.scss [hunk 5]
```diff
@@ -4,22 +4,25 @@
...
   --#{$prefix}popover-header-padding-bottom: #{$popover-header-padding-bottom}; // Boosted mod
+  --#{$prefix}popover-header-padding-top: #{$popover-header-padding-top}; // Boosted mod
   @include rfs($popover-header-font-size, --#{$prefix}popover-header-font-size);
   --#{$prefix}popover-header-line-height: #{$popover-header-line-height}; // Boosted mod
   --#{$prefix}popover-header-color: #{$popover-header-color};
   --#{$prefix}popover-header-bg: #{$popover-header-bg};
   --#{$prefix}popover-body-padding-x: #{$popover-body-padding-x};
   --#{$prefix}popover-body-padding-y: #{$popover-body-padding-y};
+  --#{$prefix}popover-body-padding-top: #{$popover-body-padding-top}; // Boosted mod
+  --#{$prefix}popover-body-padding-bottom: #{$popover-body-padding-bottom}; // Boosted mod
```
Comment (louismaximepiton) [inline scss/_popover.scss:25]: We should reorder above to have the same order appearance (l.16 and l.17)?

File: scss/_popover.scss [hunk 6]
```diff
@@ -203,15 +209,19 @@
...
+    var(--#{$prefix}popover-body-padding-y, var(--#{$prefix}popover-body-padding-top))
+    var(--#{$prefix}popover-body-padding-x)
+    var(--#{$prefix}popover-body-padding-y, var(--#{$prefix}popover-body-padding-bottom))
+    var(--#{$prefix}popover-body-padding-x); // Boosted mod: instead of padding: `var(--#{$prefix}popover-body-padding-y) var(--#{$prefix}popover-body-padding-x);`
   color: var(--#{$prefix}popover-body-color);
 }
+
+// Boosted mod
+:not(.popover-header) + .popover-body {
+  --bs-popover-body-padding-y: #{$popover-padding-y}; // Reset value to the defaut y padding popover one when there's only a body
```
Comment (louismaximepiton) [inline scss/_popover.scss:225]:
```suggestion
  --#{$prefix}popover-body-padding-y: #{$popover-padding-y}; // Reset value to the defaut y padding popover one when there's only a body
```

File: scss/_variables.scss [hunk 7]
```diff
@@ -1784,29 +1784,31 @@ $popover-border-color:              var(--#{$prefix}border-color-translucent) !d
 $popover-border-radius:             var(--#{$prefix}border-radius-lg) !default;
 $popover-inner-border-radius:       calc(#{$popover-border-radius} - #{$popover-border-width}) !default; // stylelint-disable-line function-disallowed-list
 $popover-box-shadow:                $box-shadow !default;
+$popover-padding-y:                 $spacer !default; // Boosted mod
```
Comment (louismaximepiton) [inline scss/_variables.scss:1787]: Should be reordered ? (right after max-width?)

**Review by louismaximepiton — COMMENTED** [review]
Some small changes but looks fine already

File: scss/_popover.scss (hunk 4 above)
Comment (julien-deramond) [reply scss/_popover.scss:219]: Yep let's explain it in the "CSS variables" section after the list of custom props rather than in the migration guide.
And yeah, let's remove the latest useless `var(--#{$prefix}popover-body-padding-x)`.
